File: 2264. Largest 3-Same-Digit Number in String/solution.cpp
class Solution {
public:
  string largestGoodInteger(string num) {
    string result = "";
    for (int index = 2; index < num.size(); index++){
      if (num[index - 2] == num[index - 1] && num[index - 1] == num[index]){
        if (result == ""){
          result = num.substr(index - 2, 3);
        }
        else if (stoi(num.substr(index - 2, 3)) > stoi(result)){
          result = num.substr(index - 2, 3);
        }
      }
    }
    
    return result;
  }
};

File: 2264. Largest 3-Same-Digit Number in String/solution.py
class Solution(object):
  def largestGoodInteger(self, num):
    """
    :type num: str
    :rtype: str
    """
    result = ""
    for index in range(2, len(num)):
      if num[index - 2] == num[index - 1] and num[index - 1] == num[index]:
        if result == "":
          result = num[index - 2:index + 1]
        elif int(num[index - 2:index + 1]) > int(result):
          result = num[index - 2:index + 1]
    
    return result